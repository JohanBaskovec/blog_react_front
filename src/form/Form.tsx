import React from "react";
import "./Form.scss";

interface FormProps {
    style?: React.CSSProperties;
    children?: React.ReactNode;
    name?: string;
}

export function Form(props: FormProps): JSX.Element {
    return (<form name={props.name} style={props.style} className="Form">{props.children}</form>)
}
