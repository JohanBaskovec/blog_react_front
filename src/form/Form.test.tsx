import React from "react";
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect'
import {Form} from "./Form";

describe('Form', () => {
    it("is rendered properly", () => {
        const tree = TestRenderer
            .create(<Form style={{width: '100%'}}>children</Form>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
