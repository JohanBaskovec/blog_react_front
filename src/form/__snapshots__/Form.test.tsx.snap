// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form is rendered properly 1`] = `
<form
  className="Form"
  style={
    Object {
      "width": "100%",
    }
  }
>
  children
</form>
`;
